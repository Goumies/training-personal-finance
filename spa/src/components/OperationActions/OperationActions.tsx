import React from "react";
import "./OperationActions.scss";

export interface Props {
  id: string;
  onDelete(id: string): void;
}

export const OperationActions: React.StatelessComponent<Props> = ({ id, onDelete }: Props) => {
  const onDeleteClick = () => onDelete(id);
  return (
    <div className="actions">
      <button
        className="delete mdl-button mdl-js-button mdl-button--icon"
        onClick={onDeleteClick}
        data-e2e="account-operation-delete"
      >
        <i className="material-icons">delete</i>
      </button>
    </div>
  );
};
