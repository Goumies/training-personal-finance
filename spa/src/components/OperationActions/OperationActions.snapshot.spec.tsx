import React from "react";
import renderer from "react-test-renderer";
import { OperationActions, Props } from "./OperationActions";

it("renders correctly", () => {
  const props: Props = {
    id: "0",
    onDelete: jest.fn()
  };
  const tree = renderer.create(<OperationActions {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
