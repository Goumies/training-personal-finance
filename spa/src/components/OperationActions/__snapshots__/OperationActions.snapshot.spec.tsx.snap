// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="actions"
>
  <button
    className="delete mdl-button mdl-js-button mdl-button--icon"
    data-e2e="account-operation-delete"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      delete
    </i>
  </button>
</div>
`;
